src/plugins/stake-plugins/uniswapV2/policies/masonryPolicy.ts

```diff
@@ -222,25 +222,21 @@ export const makeMasonryPolicy = (options?: MasonryPolicyOptions): StakePluginPo
           // We don't need to approve the stake pool contract for the token earned token contract
           if (allocation.allocationType === 'claim') return
           const tokenContract = makeContract(allocation.currencyCode)
-          const allowanceResponse = await multipass(p => tokenContract.connect(p).allowance(signerAddress, poolContract.address))
-          const isFullyAllowed = gte(sub(allowanceResponse._hex, toHex(allocation.nativeAmount)), '0')
-          if (!isFullyAllowed) {
-            txs.build(
-              (gasLimit =>
-                async function approvePoolContract({ signer }) {
-                  const result = await tokenContract.connect(signer).approve(poolContract.address, ethers.constants.MaxUint256, {
-                    gasLimit,
-                    gasPrice,
-                    nonce: nextNonce()
-                  })
-                  cacheTxMetadata(result.hash, nativeCurrencyCode, {
-                    name: metadataName,
-                    category: 'Expense:Fees',
-                    notes: 'Approve staking rewards pool contract'
-                  })
-                })(gasLimitAcc('50000'))
-            )
-          }
+          txs.build(
+            (gasLimit =>
+              async function approvePoolContract({ signer }) {
+                const result = await tokenContract.connect(signer).approve(poolContract.address, BigNumber.from(allocation.nativeAmount), {
```

Comment:
Remove comments on lines 217 and 218. They don't seem to make sense nor apply now with this change.